hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveCatalog.java

```diff
@@ -270,9 +275,34 @@ public void testCreateTableWithOwner() throws Exception {
       Assert.assertEquals(owner, hmsTable.getOwner());
       Map<String, String> hmsTableParams = hmsTable.getParameters();
       Assert.assertFalse(hmsTableParams.containsKey(TableProperties.HMS_TABLE_OWNER));
+      Assert.assertEquals(
+          PrincipalType.USER.name(), hmsTableParams.get(TableProperties.HMS_TABLE_OWNER_TYPE));
     } finally {
       catalog.dropTable(tableIdent);
     }
+
+    TableIdentifier tableIdent2 = TableIdentifier.of(DB_NAME, "tbl_group_owned");
+    String location2 = temp.newFolder("tbl_group_owned").toString();
+    String owner2 = "some_group_owner";
+    ImmutableMap<String, String> properties2 =
+        ImmutableMap.of(
+            TableProperties.HMS_TABLE_OWNER,
+            owner2,
+            TableProperties.HMS_TABLE_OWNER_TYPE,
+            PrincipalType.GROUP.name());
+
+    try {
+      Table table2 = catalog.createTable(tableIdent2, schema, spec, location2, properties2);
+      org.apache.hadoop.hive.metastore.api.Table hmsTable =
+          metastoreClient.getTable(DB_NAME, "tbl_group_owned");
+      Assert.assertEquals(owner2, hmsTable.getOwner());
+      Map<String, String> hmsTableParams = hmsTable.getParameters();
+      Assert.assertFalse(hmsTableParams.containsKey(TableProperties.HMS_TABLE_OWNER));
+      Assert.assertEquals(
+          PrincipalType.GROUP.name(), hmsTableParams.get(TableProperties.HMS_TABLE_OWNER_TYPE));
+    } finally {
+      catalog.dropTable(tableIdent2);
+    }
```

Comment:
I wonder if it would make sense to check the owner when no table property is given.

Reply:
Yes, it definitely would. I should add a test case with no prop specified. Though we should probably figure out what is the correct behavior in previous comments/conversations. Such as, if we can't figure out the owner, should be just set null to both owner and ownerTypes.

Reply:
Since table group ownership support is no longer relevant to this PR. This change has also been reverted.